script.js

```diff
@@ -1,12 +1,235 @@
+    idArr.push(showList[i].id);
+  }
+  for (let i = 0; i < showList.length; i++) {
+    let optionElement = document.createElement("option");
+    optionElement.value = `${idArr[i]}`;
+    optionElement.innerHTML = `${showArr[i]}`;
+    selectShowElement.appendChild(optionElement);
+  }
+};
+
+//adding event listener to choose selected show from select show option
+selectShowElement.addEventListener("change", selectedShow);
+
+function selectedShow() {
+  reset();
+  if (selectShowElement.selectedIndex === 0) {
+    mainElement.innerHTML = "";
+    searchBox("Shows");
+    displayShow();
+  } else {
+    mainElement.innerHTML = "";
+    let showId = selectShowElement.value;
+    selectEpisodeElement.innerHTML = "";
+    fetch(`https://api.tvmaze.com/shows/${showId}/episodes`)
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error("something wrong");
+        }
+        return response.json();
+      })
+      .then((data) => {
+        const allEpisodes = data;
+        dropDown(allEpisodes);
+        makePageForEpisodes(allEpisodes);
+        searchBox("Episodes");
+      })
+      .catch((error) => console.log(error));
+  }
+}
+
+// display All show on main page
+function displayShow() {
+  let showObj = getAllShows();
+  selectEpisodeElement.style.display = "none";
+  selectEpisodesLabel.style.display = "none";
+  for (let i = 0; i < showObj.length; i++) {
+    truncate;
+    showObj.sort((a, b) => (a.name > b.name ? 1 : -1));
+    let showDiv = document.createElement("div");
+    showDiv.className = "showSection";
+    if (showObj[i].image !== null) {
+      showDiv.innerHTML = `<div class=" show d-flex flex-column align-items-center mt-4">
+         <div class=" d-flex flex-column align-items-center justify-content-center">
+            <h4 class="pb-2" role="button">${showObj[i].name}</h4>
+            <img class="w-100 bd-highlight img-thumbnail" src =${showObj[i].image.medium}>
+        </div>
+        <div class="d-flex justify-content-between">
+            <div class= "summary p-4">Summary: ${showObj[i].summary}</div>
+            <section class= " rating col-sm-4 flex-column justify-content-center text-secondary p-4">
+              <div class="p-2 overflow-auto"> Genres: ${showObj[i].genres}</div>
+              <div class="p-2 overflow-auto"> Rated: ${showObj[i].rating.average}</div>
+              <div class="p-2 overflow-auto"> Status: ${showObj[i].status}</div>
+              <div class="p-2 overflow-auto"> Runtime: ${showObj[i].runtime}</div>
+            </section>
+            </div>
+      </div>
+      <hr>`;
+      mainElement.appendChild(showDiv);
+      rootElem.appendChild(mainElement);
+
+      //Adding event listener for selected show
+      showDiv.addEventListener("click", () => {
+        reset();
+        fetch(`https://api.tvmaze.com/shows/${showObj[i].id}/episodes`)
+          .then((response) => {
+            if (!response.ok) {
+              throw new Error("something wrong");
```

Comment:
It's great that you added error handling, but can you think of a way to make the message more informative? There might be some more details in the `response` object about what went wrong.